rust-connectors/sources/postgres/src/connect.rs

```diff
@@ -0,0 +1,225 @@
+use crate::convert::convert_replication_event;
+use crate::{Error, PgConnectorOpt};
+use fluvio::dataplane::smartstream::SmartStreamInput;
+use fluvio::metadata::smartmodule::SmartModuleSpec;
+use fluvio::metadata::topic::TopicSpec;
+use fluvio::{Fluvio, Offset, TopicProducer};
+use fluvio_model_postgres::{Column, LogicalReplicationMessage, ReplicationEvent};
+use fluvio_smartengine::SmartStream;
+use once_cell::sync::Lazy;
+use postgres_protocol::message::backend::{
+    LogicalReplicationMessage as PgReplication, ReplicationMessage,
+};
+use std::collections::BTreeMap;
+use std::pin::Pin;
+use std::time::{Duration, SystemTime, UNIX_EPOCH};
+use tokio_postgres::config::ReplicationMode;
+use tokio_postgres::replication::LogicalReplicationStream;
+use tokio_postgres::types::PgLsn;
+use tokio_postgres::{Client, NoTls};
+use tokio_stream::StreamExt;
+
+const TIME_SEC_CONVERSION: u64 = 946_684_800;
+static EPOCH: Lazy<SystemTime> =
+    Lazy::new(|| UNIX_EPOCH + Duration::from_secs(TIME_SEC_CONVERSION));
+
+/// A Fluvio connector for Postgres CDC.
+pub struct PgConnector {
+    /// The connector configuration.
+    config: PgConnectorOpt,
+    /// The Postgres client for streaming replication changes.
+    pg_client: Client,
+    /// The Fluvio producer for recording change events.
+    producer: TopicProducer,
+    /// The current Log Sequence Number (offset) to read in the replication stream.
+    lsn: Option<PgLsn>,
+    /// Caches the schema for each new table we see, grouped by relation_id
+    relations: BTreeMap<u32, Vec<Column>>,
+    /// The user-given SmartStream to apply to this connector's stream
+    smartstream: Option<Box<dyn SmartStream>>,
+}
+
+impl PgConnector {
+    pub async fn new(config: PgConnectorOpt) -> eyre::Result<Self> {
+        tracing::info!("Initializing PgConnector");
+        let fluvio = Fluvio::connect().await?;
+        tracing::info!("Connected to Fluvio");
+
+        let admin = fluvio.admin().await;
+
+        let smartstream: Option<Box<dyn SmartStream>> = match config.common.smarstream_module() {
```

Comment:
Make use of [`CommonSourceOpt::create_producer`](https://github.com/infinyon/fluvio-connectors/blob/2b711579f18ebcb9c91eaef647d5ef09f4751063/rust-connectors/common/src/opt.rs#L47).

Reply:
We also can create a create_consumer there